### PR TITLE
Add setup-toolchain.sh script to configure the master version of rustc,

### PR DESCRIPTION
**Pull request by jsgf**
and update CONTRIBUTING.md accordingly.

### CODE REVIEW

**Comment by jsgf** [comment]
Not sure if this is actually correct, but it unblocked me. I stole the logic from .travis.yml, so maybe it should just use this script.

File: setup-toolchain.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+# Set up the appropriate rustc toolchain
+
+cargo install rustup-toolchain-install-master --debug || echo "rustup-toolchain-install-master already installed"
+RUSTC_HASH=$(git ls-remote https://github.com/rust-lang/rust.git master | awk '{print $1}')
+rustup-toolchain-install-master -f -n master $RUSTC_HASH
+rustup override set master
```
Comment (Manishearth) [inline setup-toolchain.sh]: newline at eof please

**Comment by Manishearth** [comment]
LGTM, feel free to make travis use it

**Comment by phansch** [comment]
Failure should be fixed with #3736, going to retrigger the build

**Comment by phansch** [comment]
I think you'll also have to change 

https://github.com/rust-lang/rust-clippy/blob/3bda548f81bc268a2e9813ce9168d2e40e8a11bd/ci/base-tests.sh#L32

and

https://github.com/rust-lang/rust-clippy/blob/3bda548f81bc268a2e9813ce9168d2e40e8a11bd/ci/base-tests.sh#L52

to use `override set nightly/master` because the override takes priority over the default toolchain.

**Comment by jsgf** [comment]
OK, updated.

**Comment by Manishearth** [comment]
thanks!

**Pull request merged**